mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```
@@ -824,16 +892,129 @@ class MatMulConverter final : public OpConversionPattern<tosa::MatMulOp> {
     }
   }
 
+  // Helper to extract scale and matrix from a mul operation
+  std::pair<Value, Value> tryExtractScaleAndMatrix(Value mulInput1,
```

nit: use FailureOr instead of returning null pointers

---

Done.